runtime/src/main/java/org/corfudb/runtime/view/workflows/WorkflowRequest.java

```diff
@@ -63,8 +68,16 @@ protected ManagementClient getOrchestrator(Layout layout) {
             throw new WorkflowException("getOrchestrator: no available orchestrators " + layout);
         }
 
+        // Randomly select an orchestrator, just in-case the unresponsive servers
+        // doesn't reflect the current state of the system
+        int ind = ThreadLocalRandom.current()
+                .nextInt(activeLayoutServers.size()) % activeLayoutServers.size();
+
+        log.info("getOrchestrator: orchestrator selected {}, layout {}",
+                activeLayoutServers.get(ind), layout);
+
```

Comment:
will we consider verifying in advance if the selected node is responsive?

Reply:
Yea, I think as the number of unresponsive nodes increase, selecting one at random might not be the best strategy. @medhavidhawan  suggests that we ping.

Reply:
done.